src/vcpkg/tools.cpp

```diff
@@ -62,82 +63,114 @@ namespace vcpkg
         return std::array<int, 3>{*d1.get(), *d2.get(), *d3.get()};
     }
 
-    static Optional<ToolData> parse_tool_data_from_xml(StringView XML, StringView XML_PATH, StringView tool)
+    ExpectedL<std::vector<ToolDataEntry>> parse_tool_data(StringView contents, StringView origin)
     {
+        auto as_json = Json::parse(contents, origin);
+        if (!as_json)
+        {
+            return as_json.error();
+        }
+        auto as_value = std::move(as_json).value(VCPKG_LINE_INFO).value;
```

Comment:
This should have error handling like that in `Json::parse_object` rather than crashing if it can't be interpreted as a value here.